doc/source/scripts/multi_tetrode_example.py

```diff
@@ -0,0 +1,103 @@
+"""
+Example for usecases.rst
+"""
+
+from itertools import cycle
+import numpy as np
+from quantities import ms, mV, kHz
+import matplotlib.pyplot as plt
+from neo import Block, Segment, View, Group, SpikeTrain, AnalogSignal
+
+store_signals = False
+
+block = Block(name="probe data", tetrode_ids=["Tetrode #1", "Tetrode #2"])
+block.segments = [Segment(name="trial #1", index=0),
+                  Segment(name="trial #2", index=1),
+                  Segment(name="trial #3", index=2)]
+
+n_units = {
+    "Tetrode #1": 2,
+    "Tetrode #2": 5
+}
+
+# Create a group for each neuron, annotate each group with the tetrode from which it was recorded
+groups = []
+counter = 0
+for tetrode_id, n in n_units.items():
+    groups.extend(
+        [Group(name=f"neuron #{counter + i + 1}", tetrode_id=tetrode_id)
+         for i in range(n)]
+    )
+    counter += n
+block.groups.extend(groups)
+
+iter_group = cycle(groups)
+
+# Create dummy data, one segment at a time
+for segment in block.segments:
+    segment.block = block
+
+    # create two 4-channel AnalogSignals with dummy data
+    signals = {
+        "Tetrode #1": AnalogSignal(np.random.rand(1000, 4) * mV, sampling_rate=10 * kHz, tetrode_id="Tetrode #1"),
+        "Tetrode #2": AnalogSignal(np.random.rand(1000, 4) * mV, sampling_rate=10 * kHz, tetrode_id="Tetrode #2")
+    }
+    if store_signals:
+        segment.analogsignals.extend(signals.values())
+        for signal in signals:
+            signal.segment = segment
+
+    # create spike trains with dummy data
+    # we will pretend the spikes have been extracted from the dummy signal
+    for tetrode_id in ("Tetrode #1", "Tetrode #2"):
+        for i in range(n_units[tetrode_id]):
+            spiketrain = SpikeTrain(np.random.uniform(0, 100, size=30) * ms, t_stop=100 * ms)
+            # assign each spiketrain to the appropriate segment
+            segment.spiketrains.append(spiketrain)
+            spiketrain.segment = segment
+            # assign each spiketrain to a given neuron
+            current_group = next(iter_group)
```

Comment:
I like this example.
Maybe the use of next.cycle is a bit confusion for a python beginner.
Maybe an old school and less elegant indexing would be more explicit.
No ?

Reply:
yes, but I propose to leave it for now since we plan to rewrite the documentation (and examples) anyway.